docs/react/guides/advanced-ssr.md

```diff
@@ -34,19 +34,37 @@ The first step of any React Query setup is always to create a `queryClient` and
 import { useState } from 'react'
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
 
+function makeQueryClient() {
+  return new QueryClient({
+    defaultOptions: {
+      queries: {
+        // With SSR, we usually want to set some default staleTime
+        // above 0 to avoid refetching immediately on the client
+        staleTime: 60 * 1000,
+      },
+    },
+  })
+}
+
+let browserQueryClient: QueryClient | undefined = undefined
+
+function getQueryClient() {
+  if (typeof window === 'undefined') {
+    // Server: always make a new query client
+    return makeQueryClient()
+  } else {
+    // Browser: make a new query client if we don't already have one
+    // This is very important so we don't re-make a new client if React
+    // supsends during the initial render
```

Comment:
I guess that's why a `Suspense` boundary is a good idea - so that you can create a `queryClient` in React State outside of it, and it won't be thrown away. I don't mind the `makeQueryClient()`function - it's good that it uses a singleton on the client and a new one every time on the server 👍 . I would still probably mention somewhere that this is only needed if you don't have an explicit suspense boundary further down the tree...